File: ch33p848map/p707p723map/MyOverlay.m

#import "MyOverlay.h"


@implementation MyOverlay

@synthesize boundingMapRect, coordinate, path;


- (id) initWithRect: (MKMapRect) rect {
    self = [super init];
    if (self) {
        self->boundingMapRect = rect;
    }
    return self;
}

- (CLLocationCoordinate2D) coordinate {
    MKMapPoint pt = MKMapPointMake(MKMapRectGetMidX(self->boundingMapRect), MKMapRectGetMidY(self->boundingMapRect));
    return MKCoordinateForMapPoint(pt);
}


@end

File: iOS10bookexamples/bk2ch21p744maps/ch34p1008maps/MyOverlay.swift


import UIKit
import MapKit

class MyOverlay : NSObject, MKOverlay {
    
    var coordinate : CLLocationCoordinate2D {
        get {
            let pt = MKMapPointMake(MKMapRectGetMidX(self.boundingMapRect), MKMapRectGetMidY(self.boundingMapRect))
            return MKCoordinateForMapPoint(pt)
        }
    }
    
    var boundingMapRect : MKMapRect
    var path : UIBezierPath!
    
    init(rect:MKMapRect) {
        self.boundingMapRect = rect
        super.init()
    }
    
}
